.github/actions/ct-test/action.yml

```diff
@@ -0,0 +1,27 @@
+# SPDX-License-Identifier: Apache-2.0
+
+name: CT Test
+description: Builds the library with given flags and runs Valgrind-based constant-time tests
+
+inputs:
+  cflags:
+    description: CFLAGS to pass to compilation
+    default: ""
+
+runs:
+  using: composite
+  steps:
+      - shell: ${{ env.SHELL }}
+        run: |
+          make clean
+          OPT=1 CFLAGS="-DENABLE_CT_TESTING -DNTESTS=50 ${{ inputs.cflags }}" make func
+          valgrind --error-exitcode=1 --track-origins=yes ./test/build/mlkem512/bin/test_mlkem512
```

Comment:
Nit: You could simplify this by using `tests func --exec-wrapper ...` I think.

Reply:
Trying that now.